tests/test_indexes.py

```diff
@@ -12,23 +12,32 @@
 import numpy as np
 from numpy import typing as npt
 import pandas as pd
-from pandas.core.indexes.numeric import NumericIndex
 from typing_extensions import (
     Never,
     assert_type,
 )
 
 from pandas._typing import Scalar
 
-if TYPE_CHECKING:
-    from pandas._typing import IndexIterScalar
-
 from tests import (
+    PD_LTE_15,
     TYPE_CHECKING_INVALID_USAGE,
     check,
     pytest_warns_bounded,
 )
 
+if TYPE_CHECKING:
+    from pandas.core.indexes.numeric import NumericIndex
+
+    from pandas._typing import IndexIterScalar
+else:
+    if not PD_LTE_15:
+        from typing_extensions import TypeAlias
+
+        NumericIndex: TypeAlias = pd.Index
```

Comment:
`from pandas import Index as NumericIndex` could be slightly simpler

Reply:
Good idea.  Fixed in commit https://github.com/pandas-dev/pandas-stubs/pull/528/commits/600ab56f654a20a7633847d766a13eca3faa32ab